src/emscripten-optimizer/simple_ast.h

```diff
@@ -1006,7 +1006,7 @@ struct JSPrinter {
     if (childPrecedence < parentPrecedence) return false; //          definitely cool
     // equal precedence, so associativity (rtl/ltr) is what matters
     // (except for some exceptions, where multiple operators can combine into confusion)
-    if (parent[0] == UNARY_PREFIX) {
+    if (parent->isArray() && parent[0] == UNARY_PREFIX) {
```

Comment:
I wasn't able to find a reproducer after a quick check just now, but if I remember correctly at one point during development I had a situation where `parent` was either an `Assign` or an `AssignName` node, neither of which are Arrays.